hyperspy/io_plugins/emd.py

```diff
@@ -0,0 +1,392 @@
+        # Write microscope:
+        microscope_group = emd_file.create_group('microscope')
+        for key, value in self.microscope.iteritems():
+            microscope_group.attrs[key] = value
+        # Write sample:
+        sample_group = emd_file.create_group('sample')
+        for key, value in self.sample.iteritems():
+            sample_group.attrs[key] = value
+        # Write comments:
+        comments_group = emd_file.create_group('comments')
+        for key, value in self.comments.iteritems():
+            comments_group.attrs[key] = value
+        # Write signals:
+        signal_group = emd_file.create_group('signals')
+        for signal in self.signals.values():
+            self._write_signal_to_group(signal_group, signal)
+        # Close file and return EMD object:
+        emd_file.close()
+
+    def print_info(self):
+        """Print all relevant information about the EMD instance.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        None
+
+        """
+        self._log.debug('Calling print_info')
+        print '\nUser:\n--------------------'
+        for key, value in self.user.iteritems():
+            print '{}:'.format(key).ljust(15), value
+        print '--------------------\n\nMicroscope:\n--------------------'
+        for key, value in self.microscope.iteritems():
+            print '{}:'.format(key).ljust(15), value
+        print '--------------------\n\nSample:\n--------------------'
+        for key, value in self.sample.iteritems():
+            print '{}:'.format(key).ljust(15), value
+        print '--------------------\n\nComments:\n--------------------'
+        for key, value in self.comments.iteritems():
+            print '{}:'.format(key).ljust(15), value
+        print '--------------------\n\nData:\n--------------------'
+        for key, value in self.signals.iteritems():
+            print '{}:'.format(key).ljust(15), value
+            print value.metadata.Signal
+        print '--------------------\n'
+
+
+def file_reader(filename, load_to_memory=True, print_info=False, **kwds):
+    emd = EMD.load_from_emd(filename, load_to_memory)
+    if print_info:
+        emd.print_info()
+    dictionaries = []
+    for signal in emd.signals.values():
+        axes = []
+        for i in range(len(signal.data.shape)):
+            axes.append({'size': signal.axes_manager[i].size,
+                         'index_in_array': signal.axes_manager[i].index_in_array,
+                         'name': signal.axes_manager[i].name,
+                         'scale': signal.axes_manager[i].scale,
+                         'offset': signal.axes_manager[i].offset,
+                         'units': signal.axes_manager[i].units})
+        dictionary = {'data': signal.data,
+                      'axes': axes,
+                      'metadata': signal.metadata.as_dictionary(),
+                      'original_metadata': signal.original_metadata.as_dictionary()}

```

Comment:
I didn't know about this handy function when I wrote this code and I forgot  to change it here when you mentioned it on my SEMPER plugin. It's changed now and looks much cleaner :-)!